### PR TITLE
feat: add telegraf container

### PR DESCRIPTION
**Pull request by kiragoo**

### CODE REVIEW

**Comment by Rory-Z** [comment]
Maybe we should add a check for `telegraf` to controller_suite

**Comment by kiragoo** [comment]
> 

Get it.i'll add some test for telegraf.

File: pkg/service/generator.go [hunk 1]
```diff
@@ -99,6 +106,63 @@ func generateStatefulSetDef(emqx v1beta1.Emqx) *appsv1.StatefulSet {
...
+	logName := fmt.Sprintf("%s-%s", emqx.GetName(), "log")
+
+	containerForTelegraf := corev1.Container{
+		Name:            "telegraf",
+		Image:           telegrafTemplate.Image,
+		ImagePullPolicy: "IfNotPresent",
+		Resources:       emqx.GetTelegrafTemplate().Resources,
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      telegrafConfName,
```
Comment (Rory-Z) [inline pkg/service/generator.go:122]:
```suggestion
				Name:      "telegraf",
```

**Comment by Rory-Z** [comment]
We still need more test cases

File: pkg/service/generator.go (hunk 1 above)
Comment (kiragoo) [reply pkg/service/generator.go:122]: The `Name` is generated by the `fmt.Sprintf()`, which is related with `EMQX` and used by the `VolumeMouts`,`ConfigMap` and `Volumes`.

Comment (Rory-Z) [reply pkg/service/generator.go:122]: My bad, misread

File: controllers_suite/telegraf_test.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+// http://onsi.github.io/ginkgo/ to learn more about Ginkgo.
+var _ = Describe("", func() {
+	Context("Check telegraf Conf", func() {
+		It("Check loaded plugins", func() {
+			for _, emqx := range emqxList() {
+				check_telegraf(emqx)
+			}
+		})
+
+		It("Check update plugins", func() {
```
Comment (Rory-Z) [inline controllers_suite/telegraf_test.go:27]: Delete this block

Comment (kiragoo) [reply controllers_suite/telegraf_test.go:27]: my bad.

**Review by Rory-Z — APPROVED** [review]

**Pull request merged**